test/test-connector-swagger-client.js

```diff
@@ -0,0 +1,74 @@
+// Copyright IBM Corp. 2016,2017. All Rights Reserved.
+// Node module: loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+'use strict';
+
+var should = require('should');
+var loopback = require('loopback');
+var http = require('http');
+
+function getMySwaggerClient() {
+  return {
+    execute: function(requestObject) {
+      if (requestObject.method === 'GET') {
+        http.get(requestObject.url, function(response) {
+          var body = '';
+          response.on('data', function(d) {
+            body += d;
+          });
+          response.on('end', function() {
+            var parsed = JSON.parse(body);
+            requestObject.on.response(parsed);
+          });
+        }).on('error', function(e) {
+          requestObject.on.error('{\'error\': ' +
+            '\'Test Client Error in getting response\'}');
+        });
+      }
+    },
+  };
+};
+
+describe('swagger client', function() {
+  describe('swagger connector validation with and w/o custom swagger client',
+    function() {
+      it('invoke a get call in pet service using custom swagger client',
```

Comment:
The test code is IMO too complex. I am proposing to move the first test into `test/test-connector.js` [here](https://github.com/strongloop/loopback-connector-swagger/blob/6aae164bddaa9c14f4ed2b9b8105c3a9980e1219/test/test-connector.js#L145-L146).

Here is a mock-up of what I am envisioning:

```js
it('supports custom swagger client', done => {
  const swaggerClient = {
    execute: r => r.on.response({id: 'custom'});
  };
  const ds = createDataSource('test/fixtures/petStore.json', {swaggerClient});
  const PetService = ds.createModel('PetService', {});

  PetService.on('connected', () => {
    PetService.getPetById({petId: 7}, function(err, res) {
      if (err) return done(err);
      should(res).containDeep({id: 'custom'});
      done();
    });
  });
});
```

(The example assumes that `createDataSource` is refactored to accept an `options` object instead of a list of positional arguments).